tfx/orchestration/experimental/kubernetes/kubernetes_dag_runner.py

```diff
@@ -0,0 +1,378 @@
+from tfx.orchestration import metadata
+from tfx.orchestration import pipeline as tfx_pipeline
+from tfx.orchestration import tfx_runner
+from tfx.orchestration.config import base_component_config
+from tfx.orchestration.config import config_utils
+from tfx.orchestration.config import pipeline_config
+from tfx.orchestration.kubeflow import node_wrapper
+from tfx.orchestration.kubeflow import utils
+from tfx.orchestration.launcher import base_component_launcher
+from tfx.orchestration.launcher import in_process_component_launcher
+from tfx.orchestration.launcher import kubernetes_component_launcher
+from tfx.utils import json_utils, kube_utils
+from google.protobuf import json_format
+from kubernetes import client
+import json
+
+_CONTAINER_COMMAND = [
+    'python', '/tfx-src/tfx/orchestration/experimental/kubernetes/container_entrypoint.py'
+]
+
+_DRIVER_COMMAND = [
+    'python', '/tfx-src/tfx/orchestration/experimental/kubernetes/driver_container_entrypoint.py'
+]
+
+_WRAPPER_SUFFIX = 'Wrapper'
+
+_TFX_IMAGE = 'gcr.io/tfx-eric/tfx-dev'
+
+
+def _sanitize_pod_name(pod_name: Text) -> Text:
+  pod_name = re.sub(r'[^a-z0-9-]', '-', pod_name.lower())
+  pod_name = re.sub(r'^[-]+', '', pod_name)
+  return re.sub(r'[-]+', '-', pod_name)
+
+
+def is_inside_cluster() -> bool:
+  """Determines if kubernetes dag runner is executed from within a cluster.
+  Can be pacthed for testing purpose.
+  """
+  return kube_utils.is_inside_cluster()
+
+
+def get_default_kubernetes_metadata_config(
+) -> metadata_store_pb2.ConnectionConfig:
+  """Returns the default metadata connection config for a kubernetes cluster.
+
+  Returns:
+    A config proto that will be serialized as JSON and passed to the running
+    container so the TFX component driver is able to communicate with MLMD in
+    a kubernetes cluster.
+  """
+  connection_config = metadata_store_pb2.ConnectionConfig()
+  connection_config.mysql.host = 'mysql'
+  connection_config.mysql.port = 3306
+  connection_config.mysql.database = 'mysql'
+  connection_config.mysql.user = 'root'
+  connection_config.mysql.password = ''
+  return connection_config
+
+
+def _wrap_container_component(
+    component: base_node.BaseNode,
+    component_launcher_class:
+        Type[base_component_launcher.BaseComponentLauncher],
+    component_config: Optional[base_component_config.BaseComponentConfig],
+    pipeline: tfx_pipeline.Pipeline,
+) -> base_node.BaseNode:
+  """Wrapper for container component.
+
+  Args:
+  component: Component to be executed.
+  component_launcher_class: The class of the launcher to launch the
+    component.
+  component_config: component config to launch the component.
+  pipeline: Logical pipeline that contains pipeline related information.
+  """
+
+  # Reference: tfx.orchestration.kubeflow.base_component
+  component_launcher_class_path = '.'.join([
+      component_launcher_class.__module__, component_launcher_class.__name__
+  ])
+
+  serialized_component = utils.replace_placeholder(
```

Comment:
Please add a comment about the serialization scheme, and add a TODO(ccy) to look at a different scheme once available.